### PR TITLE
(GH-578) etc_dir should be configurable

### PR DESCRIPTION
**Pull request by Phil-Friderici**

### CODE REVIEW

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -26,5 +26,6 @@
     :ipaddress   => '127.0.0.1',
     :kernel      => 'Linux',
     :osfamily    => 'RedHat',
+    :fqdn        => 'testfqdn',
```
Comment (ghoneycutt) [inline spec/spec_helper.rb]: your fqdn isnt fully qualified, perhaps `testfqdn.example.com`

**Comment by Phil-Friderici** [comment]
Wow, this small parameter does touch a lot of resources.

The sensu::check class does not source in $etc_dir from sensu main class. It does use it's own (same) default values. Should I change senus::check to also source values from main class ?

File: spec/spec_helper.rb (hunk 1 above)
Comment (Phil-Friderici) [inline spec/spec_helper.rb]: fixed

**Comment by Phil-Friderici** [comment]
will rebase

**Comment by Phil-Friderici** [comment]
rebased & ready for review

File: manifests/init.pp [hunk 2]
```diff
@@ -22,6 +22,12 @@
 #   Default: undef
 #   Valid values: sensu_gem, apt, aptitude, yum
 #
+# [*sensu_etc_dir*]
+#   String. Absolute path to the sensu etc directory.
+#   Default: '/etc/sensu' on RedHat and Debian
+#            'C:/opt/sensu' on windows
+#            undef on all other platforms
```
Comment (ghoneycutt) [inline manifests/init.pp]: what happens if this is undef? Maybe the default should be `/etc/sensu` ?

Comment (Phil-Friderici) [inline manifests/init.pp]: That would make sense in most cases I guess.

**Comment by Phil-Friderici** [comment]
@tuxmea Let's add support for new OSfamilies in specific PRs/commits. Guess there is more work to do than just adding different paths for $sensu_etc_dir.

File: manifests/init.pp (hunk 2 above)
Comment (Phil-Friderici) [inline manifests/init.pp]: changed the behaviour to default to '/etc/sensu'

File: spec/classes/sensu_init_spec.rb [hunk 3]
```diff
@@ -30,6 +136,106 @@
       }
     end
 
+    # FIXME: The following resource checks are only testing $sensu_etc_dir specific values
```
Comment (ghoneycutt) [inline spec/classes/sensu_init_spec.rb]: what else should they test?

Comment (Phil-Friderici) [inline spec/classes/sensu_init_spec.rb]: These tests only checks for the $sensu_etc_dir specific details. They don't test the other file resource details that should be tested as well. Eg:

``` ruby
      it do
        should contain_file('C:/opt/sensu/bin/sensu-client.xml').with({
          'ensure'  => 'file',
          'content' => content,
          ... and other parameters....
        })
      end
```
There is still a lot of test to add for all the managed resources and functionality. But this is out of scope for this PR.

**Comment by Phil-Friderici** [comment]
will rebase

**Comment by Phil-Friderici** [comment]
rebased

**Comment by ghoneycutt** [comment]
Sorry to ask you to rebase again, though we've implemented data types and all PR's will need to be rebased.

**Comment by Phil-Friderici** [comment]
et voilà, rebased for a good thing ;)

**Comment by ghoneycutt** [comment]
Solves #578

**Comment by ghoneycutt** [comment]
Thank you! Released in v2.28.0
